src/measure_theory/interval_integral.lean

```diff
@@ -1345,4 +1345,128 @@ theorem integral_deriv_eq_sub (hderiv : ∀ x ∈ interval a b, differentiable_a
 integral_eq_sub_of_has_deriv_at (λ x hx, (hderiv x hx).has_deriv_at) hcont'
   (measurable_deriv f).ae_measurable
 
+/-!
+### Integration by parts
+ -/
+
+lemma self_mem_ae_restrict {s : set ℝ} (hs : is_measurable s):
+  s ∈ (measure.restrict measure_space.volume s).ae :=
```

Comment:
This lemma should move to an earlier file -- for example, I think it would be at home close to `measure_theory.ae_restrict_eq` in the file `measure_theory.measure_space`, and perhaps could be deduced from it.  Like that lemma, I think it should be stated for an arbitrary `(μ : measure α)` rather than just `volume`.